lib/libvmod_blob/hex.c

```diff
@@ -161,6 +176,17 @@ hex_decode(const enum encoding dec, char *restrict const buf,
 		}
 		extranib = *s;
 	}
+
+	/* XXX: no extra nible check for the last string? */
```

Comment:
No. If there is an extranib, and when neither of *s and *(s+1) are '\0' the last time through the while loop, then after s+=2 in the while loop, *s!='\0' and *(s+1)=='\0'. Then since if(extranib) is true, s++ gets executed. After which *s=='\0', the string is read all the way through, and the while loop is not entered.

Reply:
Just want to add that there are tests of odd-length hex encodings in the VTCs, all passing.